packages/expo-cli/src/commands/eject/Eject.ts

```diff
@@ -107,32 +86,101 @@ export async function ejectAsync(projectRoot: string, options: EjectAsyncOptions
+  if (ConfigWarningAggregator.hasWarningsIOS()) {
+    applyingIOSConfigStep.stopAndPersist({
+      symbol: '⚠️ ',
+      text: chalk.red('iOS configuration applied with warnings that should be fixed:'),
+    });
+    logConfigWarningsIOS();
+  } else {
+    applyingIOSConfigStep.succeed('All project configuration applied to iOS project');
   }
+  log.newLine();
+}
 
-  const outputMap: DependenciesMap = {};
-  if (!dependencies) return outputMap;
+async function installPodsAsync(projectRoot: string) {
+  log.newLine();
+  let step = logNewSection('Installing CocoaPods.');
+  if (process.platform !== 'darwin') {
+    step.succeed('Skipped installing CocoaPods because operating system is not on macOS.');
+    return false;
+  }
+  const packageManager = new PackageManager.CocoaPodsPackageManager({
+    cwd: path.join(projectRoot, 'ios'),
+    log,
+    silent: !process.env.EXPO_DEBUG,
+  });
 
-  for (const key of Object.keys(dependencies)) {
-    const value = dependencies[key];
-    if (typeof value === 'string') {
-      outputMap[key] = value;
-    } else {
-      throw new Error(
-        `Dependency for key \`${key}\` should be a \`string\`, instead got: \`{ ${key}: ${JSON.stringify(
-          value
-        )} }\``
-      );
+  if (!(await packageManager.isCLIInstalledAsync())) {
+    try {
+      // prompt user -- do you want to install cocoapods right now?
+      step.text = 'CocoaPods CLI not found in your PATH, installing it now.';
+      step.render();
+      await packageManager.installCLIAsync();
+      step.succeed('Installed CocoaPods CLI');
+      step = logNewSection('Running `pod install` in the `ios` directory.');
+    } catch (e) {
+      step.stopAndPersist({
+        symbol: '⚠️ ',
+        text: chalk.red(
+          'Unable to install the CocoaPods CLI. Continuing with ejecting, you can install CocoaPods afterwards.'
+        ),
+      });
+      if (e.message) {
+        log(`- ${e.message}`);
+      }
+      return false;
     }
   }
-  return outputMap;
+
+  try {
+    await packageManager.installAsync();
+    step.succeed('Installed pods and initialized Xcode workspace.');
+    return true;
+  } catch (e) {
+    step.stopAndPersist({
+      symbol: '⚠️ ',
+      text: chalk.red(
+        'Something when wrong running `pod install` in the `ios` directory. Continuing with ejecting, you can debug this afterwards.'
+      ),
+    });
+    if (e.message) {
+      log(`- ${e.message}`);
+    }
+    return false;
+  }
+}
+
+async function installNodeModulesAsync(projectRoot: string) {
+  let installingDependenciesStep = logNewSection('Installing JavaScript dependencies.');
+  await fse.remove('node_modules');
+  const packageManager = PackageManager.createForProject(projectRoot, { log, silent: true });
+  try {
+    await packageManager.installAsync();
+    installingDependenciesStep.succeed('Installed JavaScript dependencies.');
+  } catch (e) {
+    installingDependenciesStep.fail(
```

Comment:
Maybe add a comment here as to why it's ok to keep going even if installing the packages failed